src/Freedom/UI.purs

```diff
@@ -98,6 +98,39 @@ data VElement state
 -- | The type of virtual node.
 data VNode state = VNode String (VElement state)
 
+instance eqVNode :: Eq (VNode state) where
+  eq (VNode str el) (VNode str2 el2) = str == str2 && el == el2
+
+instance eqVElement :: Eq (VElement state) where
+  eq (Text str1) (Text str2) = str1 == str2
+  eq (Text _) _ = false
+  eq _ (Text _) = false
+  eq (Element b1 vobj1) (Element b2 vobj2) = b1 == b2 && (equalVObject vobj1 vobj2)
+
+equalVObject :: forall state. VObject state -> VObject state -> Boolean
+equalVObject vObj1 vObj2 = 
+    vObj1.tagName == vObj2.tagName 
+    && vObj1.fingerprint == vObj2.fingerprint
+    && vObj1.props == vObj2.props
+    && vObj1.children == vObj2.children
+
+instance showVNode :: Show (VNode state) where
+  show (VNode str el) = "Vnode: " <> str <> " " <> show el
```

Comment:
Generally, `Show` instances are written the following:

```purescript
"(VNode " <> show str <> " " <> show el <> ")"
```

And, I want you to use `k` instead `str`, because It's a key of `VNode` and `k` is used in this file.

Reply:
Good point.